routes/project/location/postgres.rb

```diff
@@ -84,16 +87,15 @@ class Clover
           request_body_params = Validation.validate_request_body(json_params, required_parameters)
           parsed_cidr = Validation.validate_cidr(request_body_params["cidr"])
 
-          DB.transaction do
+          firewall_rule = DB.transaction do
             PostgresFirewallRule.create_with_id(
               postgres_resource_id: pg.id,
               cidr: parsed_cidr.to_s
-            )
-            pg.incr_update_firewall_rules
+            ).tap { pg.incr_update_firewall_rules }
```

Comment:
paging @jeremyevans : this unusual use of tap I do once in a while,  because `create_with_id` is not powerful enough to allow subsequent incr of semaphores. Otherwise awkward and error-prone binding of symbols has to be done. But maybe the unusual pattern of using `tap` is a lesser defect than making a special device for our ubid'd model creations.

Reply:
Possibly create_with_id ought to take a block that merely gets run after.

Reply:
Having `create_with_id` take a block seems like a good approach to me.

Reply:
perhaps in this case I could actually reorder the increment before, though in others...with `assemble`...the strand is not yet created, so there's a foreign key problem. Maybe I'll look to tweak this one to get rid of tap.

Reply:
reordered it. it cost me a line, but I think is less subtle (usually tap uses the received object)